de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/property/BeanPropertyType.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2020 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.concept.types.property;
+
+public enum BeanPropertyType {
+	BOOLEAN("boolean"),
+	STRING("string"),
+	INT("int"),
+	FLOAT("float"),
+	ENUM("enum"),
+	RESOURCE("resource"),
+	REFERENCE("reference"),
+	EREFERENCE("ereference"),
+	COMPOSED("composed");
+
+	private final String type;
+
+	BeanPropertyType(String type) {
+		this.type = type;
+	}
+
+	public String getType() {
+		return type;
+	}
+
+	/**
+	 * Get a BeanPropertyType from a String s
+	 * @param s String
+	 * @return the BeanPropertyType
+	 */
+	public static BeanPropertyType fromString(String s) {
```

Comment:
There should probably be a BeanPropertyTypeTest where we test the fromString method.